pypesto/objective/amici_objective.py

```diff
@@ -571,6 +448,45 @@ def set_parameter_scale(self, condition_ix):
 
         self.amici_model.setParameterScale(amici_scale_vector)
 
+    def apply_steadystate_guess(self, condition_ix, x):
+        """
+        This function uses the the stored steadystate as well as the
+        respective  sensitivity (if available) and parameter value to
+        approximate the steadystate at the current parameters using a
+        zero or first order taylor approximation:
+        x_ss(x') = x_ss(x) [+ dx_ss/dx(x)*(x'-x)]
+        """
+        mapping = self.mapping_par_opt_to_par_sim[condition_ix]
+        x_sim = map_par_opt_to_par_sim(mapping, self.x_ids, x)
+        x_ss_guess = []  # resets initial state by default
+        if condition_ix in self.preeq_guesses['data']:
+            guess_data = self.preeq_guesses['data'][condition_ix]
+            if guess_data['x_ss'] is not None:
+                x_ss_guess = guess_data['x_ss']
+            if guess_data['sx_ss'] is not None:
+                x_ss_guess += guess_data['sx_ss'].transpose().dot(
+                    (x_sim - guess_data['x'])
+                )
+
+        self.amici_model.setInitialStates(x_ss_guess)
+
+    def store_steadystate_guess(self, condition_ix, x, rdata):
+
+        if condition_ix not in self.preeq_guesses['data']:
+            return
+
+        preeq_guesses = self.preeq_guesses['data'][condition_ix]
+
+        # update parameter
+
+        mapping = self.mapping_par_opt_to_par_sim[condition_ix]
+        x_sim = map_par_opt_to_par_sim(mapping, self.x_ids, x)
+        preeq_guesses['x'] = x_sim
+
+        # update steadystates
+        preeq_guesses['x_ss'] = rdata['x_ss']
```

Comment:
above, x_ss is tested for None. So I guess amici writes None in here if not available?

Reply:
correct